internal/loader/typed_setting.go

```diff
@@ -118,3 +104,51 @@ func isJsonContentType(contentType *string) bool {
 	matched, _ := regexp.MatchString("^application\\/(?:[^\\/]+\\+)?json(;.*)?$", contentTypeStr)
 	return matched
 }
+
+// UnmarshalConfigMap is used for scenarios related to feature flag refresh.
+// Currently, only supports feature flag settings in JSON/YAML formats.
+func UnmarshalConfigMap(existingConfigMapSetting *map[string]string, dataOptions *acpv1.ConfigMapDataOptions) (map[string]interface{}, map[string]interface{}, error) {
+	allSettings := make(map[string]interface{})
+	existingConfigMapData := (*existingConfigMapSetting)[dataOptions.Key]
+	var typedFeatureFlags map[string]interface{}
+	var err error
+
+	switch dataOptions.Type {
+	case acpv1.Yaml:
+		err = yaml.Unmarshal([]byte(existingConfigMapData), &allSettings)
+		if err != nil {
+			return nil, nil, err
+		}
+	case acpv1.Json:
+		err = json.Unmarshal([]byte(existingConfigMapData), &allSettings)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	if featureFlagSection, ok := allSettings[FeatureManagementSectionName].(map[string]interface{}); ok {
+		typedFeatureFlags = featureFlagSection
+		delete(allSettings, FeatureManagementSectionName)
+	}
+
+	return typedFeatureFlags, allSettings, nil
```

Comment:
it's not really allSettings are returned, do you mind rename `allSettings` to `keyValueSettings`?